Realm/Realm.Weaver/Analytics/AnalyticsUtils.cs

```diff
@@ -213,6 +191,74 @@ public static string GetLanguageVersion(ModuleDefinition module, string targetFr
             }
         }
 
+        // Knowledge on unique machine Ids for different OSes
+        // obtained from https://github.com/denisbrodbeck/machineid
+        public static string GetAnonymizedUserId()
+        {
+            var id = string.Empty;
+            var currentOs = Environment.OSVersion.Platform;
+
+            try
+            {
+                if (currentOs == PlatformID.Win32S || currentOs == PlatformID.Win32Windows ||
+                    currentOs == PlatformID.Win32NT || currentOs == PlatformID.WinCE)
+                {
+                    var rk = Registry.LocalMachine;
+                    id = (string)rk.OpenSubKey("SOFTWARE\\Microsoft\\Cryptography").GetValue("MachineGuid");
```

Comment:
Change to launch process to extract the registry value, instead of using windows specific libs.